src/microbe_stage/MicrobeHUD.cs

```diff
@@ -805,20 +814,28 @@ private void UpdateHealth(float delta)
         if (delta <= 0)
             return;
 
-        var hp = 0.0f;
-        var maxHP = 100.0f;
-
+        // Update to the player's current HP, unless the player is dead
         if (stage.Player != null)
         {
             hp = stage.Player.Hitpoints;
             maxHP = stage.Player.MaxHitpoints;
         }
+        else
+        {
+            hp = 0.0f;
+        }
 
         healthBar.MaxValue = maxHP;
-        healthBar.Value = MathUtils.Lerp((float)healthBar.Value, hp, 3.0f * delta, 0.1f);
+        SmoothlyUpdateBar(healthBar, hp, delta);
         hpLabel.Text = StringUtils.FormatNumber(Mathf.Round(hp)) + " / " + StringUtils.FormatNumber(maxHP);
     }
 
+    private void SmoothlyUpdateBar(TextureProgress bar, float target, float delta)
+    {
+        var weight = (float)Math.Min(3.0f * delta + 0.2f, 1.0f);
```

Comment:
I'd put in a 0 or negative delta guard in this method and link that issue that the delta guards link.